examples/link_prediction/heterogeneous_training.py

```diff
@@ -399,6 +372,11 @@ def _training_process(
             process_start_gap_seconds=process_start_gap_seconds,
         )
 
+        train_main_loader_iter = InfiniteIterator(train_main_loader)
```

Comment:
```suggestion
        # We keep track of both the dataloader and the InfiniteIterator for it
        # so we can clean up resources from the dataloader later.
        train_main_loader_iter = InfiniteIterator(train_main_loader)
```

And add this elsewhere?